rust_src/src/lisp.rs

```diff
@@ -559,6 +560,27 @@ impl LispObject {
             v.is_pseudovector(PseudovecType::PVEC_FONT)
         })
     }
+
+    pub fn is_font_spec(self) -> bool {
```

Comment:
Instead of adding these to `LispObject`, I would create an `as_font` method on `is_vectorlike` that returns a "LispFontRef" or similar.

`is_spec`, `is_entity` etc can the be methods on that struct.